### DISCUSSION TITLE
[Runtime] Allow overriding of the `ArgvInput` configuration in the SymfonyRuntime

### DISCUSSION THREAD

**Discussion by noemi-salaun** [symfony/symfony#43309] [Q&A]
Hello,

I have a question about the new Runtime Component and the `SymfonyComponent` implementation.

Without using the Runtime, the `bin/console` file contains

```php
$input = new ArgvInput();
if (null !== $env = $input->getParameterOption(['--env', '-e'], null, true)) {
    putenv('APP_ENV='.$_SERVER['APP_ENV'] = $_ENV['APP_ENV'] = $env);
}

if ($input->hasParameterOption('--no-debug', true)) {
    putenv('APP_DEBUG='.$_SERVER['APP_DEBUG'] = $_ENV['APP_DEBUG'] = '0');
}
```

For my needs, I followed the same idea to add a new configuration option

```php
if (null !== $lang = $input->getParameterOption(['--lang'], null, true)) {
    putenv('APP_LANG='.$_SERVER['APP_LANG'] = $_ENV['APP_LANG'] = $lang);
}
```

But when using the Runtime with the `SymfonyRuntine`, all that logic is done in the `getInput()` method. My problem is that `$input` and `getInput()` are private, so I cannot properly override it to add my conf, without copy/pasting all the `SymfonyRuntime` class, just to add my 2 lines.

Would it be possible to make `getInput()` protected to allow us to complete the conf if necessary?

**Comment by Nek-** [comment]
Hello, I feel bad about what you are trying to do, but I understand there's a problem to you. Did you consider any other solution?

I think maybe the issue (it's an issue ^^, you should create one) is more that the console component does not dispatch a "starting" event or something like that. WDYT?

**Reply by noemi-salaun** [reply]
Yes I can do 
```
APP_LANG=fr ./bin/console do:something
```

instead of
```
./bin/console --lang=fr do:something
```

but in this case the env var is not documented in the `bin/console --help` command. And you can do the say the same thing for `APP_ENV`, so what are the arguments to have a `--env` option ?

You can look at this discussion https://github.com/symfony/symfony/discussions/43113 if you want the full story behind my usage of APP_LANG ;)

**Reply by faizanakram99** [reply]
IIRC, at some point symfony decided to drop `--env` option in favour of `APP_ENV` env variable but community was not happy with that decision so `--env` was kept as it is.

One can say that `--env` is kept just for legacy reasons, otherwise `APP_ENV=env ./bin/console` just works.
